### PR TITLE
DataSourceAzure: push dmesg log to KVP

### PR DESCRIPTION
**Pull request by anhvoms**
## Proposed Commit Message
DataSourceAzure: push dmesg log to KVP

Pushing dmesg log to KVP to help troubleshoot VM boot issues
## Checklist:
 - [x] My code follows the process laid out in [the documentation](https://cloudinit.readthedocs.io/en/latest/topics/hacking.html)
 - [x] I have updated or added any unit tests accordingly
 - [x] I have updated or added any documentation accordingly

### CODE REVIEW

**Review by lucasmoura — APPROVED** [review]

**Review by mitechie — APPROVED** [review]

File: cloudinit/sources/helpers/azure.py [hunk 1]
```diff
@@ -239,6 +240,15 @@ def push_log_to_kvp(file_name=CFG_BUILTIN['def_log_file']):
             "Exception when dumping log file: %s" % repr(ex),
             logger_func=LOG.warning)
 
+    LOG.debug("Dumping dmesg log to KVP")
+    try:
+        out, _ = subp.subp(['dmesg'], decode=False, capture=True)
```
Comment (blackboxsw) [inline cloudinit/sources/helpers/azure.py:245]: Thanks for this @anhvoms . Could we get some semblance of unit test coverage for this path please.

Maybe we could have a test in tests/unittests/test_reporting_hyperv.py  confirming that regardless of what "log" is being written that dmesg is also being performed? 

I'd also really appreciate if we can get a test there that raised an error from report_compressed_event during push_log_to_kvp (to assert Exception handling behavior).

**Review by blackboxsw — CHANGES_REQUESTED** [review]
Thanks @anhvoms if we can add some unit test coverage this should be good to go.

File: cloudinit/sources/helpers/azure.py (hunk 1 above)
Comment (anhvoms) [reply cloudinit/sources/helpers/azure.py:245]: Thanks for the review @blackboxsw . I have added unit tests to cover the angles you mentioned.

**Review by blackboxsw — APPROVED** [review]
Excellent @anhvoms thanks +1

**Pull request merged**